### PR TITLE
Provide kwargs for Labels.arange()

### PR DESCRIPTION
**Pull request by frostedoyster**
----
:books: Documentation preview :books:: https://equistore--242.org.readthedocs.build/en/242/

### CODE REVIEW

File: python/src/equistore/labels.py [hunk 1]
```diff
@@ -271,25 +271,81 @@ def arange(name, *args):
...
+                    "the maximum number of integer arguments accepted by "
+                    f"`Labels.arange()` is 3. {args_len} were provided"
                 )
-
-        labels = Labels(names=[name], values=np.arange(*args).reshape(-1, 1))
+            for arg in args:
+                if not np.issubdtype(type(arg), np.integer):
+                    raise ValueError(
+                        "all numbers provided to `Labels.arange()` must be integers"
+                    )
```
Comment (Luthaf) [inline python/src/equistore/labels.py:326]: Do we need to check this ourself? Shouldn't numpy do this for us?

Comment (frostedoyster) [reply python/src/equistore/labels.py:325]: No, unfortunately `np.arange` also works with floats, and it gives float ranges out in that case, which would not work for us. It works a bit like linspace with slightly different arguments in that case

Comment (frostedoyster) [reply python/src/equistore/labels.py:328]: Ok, perfect. I've added the comments

**Comment by PicoCentauri** [comment]
Before we merge this can we try to merge the changes of the tests in #235. I think rebasing here is easier than if I have to rebase #235 again.

File: python/tests/labels.py [hunk 2]
```diff
@@ -202,3 +210,11 @@ def test_arange_incorrect_arguments():
...
     with pytest.raises(ValueError, match="integer"):
         Labels.arange("dummy", 0, 5, 0.2)
+    with pytest.raises(ValueError, match="argument"):
+        Labels.arange("dummy", random=10)
+    with pytest.raises(EquistoreError, match="label name"):
+        Labels.arange(0, 1, 2)
+    with pytest.raises(ValueError, match="integer"):
+        Labels.arange(name="dummy", start=0.0, stop=1.0, step=2)
+    with pytest.raises(ValueError, match="integer"):
+        Labels.arange("dummy", start=0, stop=0.2)
```
Comment (Luthaf) [inline python/tests/labels.py:220]: Could you update these to use the full error message?

Comment (frostedoyster) [reply python/tests/labels.py]: Sure, done

**Comment by Luthaf** [comment]
The history on this branch is weird ... Did you rebase and then merge? 

Could you squash all commits to a single one to make sure we understand what's being changed here?

**Comment by frostedoyster** [comment]
Yes... How do I do that?

**Comment by Luthaf** [comment]
> How do I do that?

This is a good introduction: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history. You'll have to fix any potential conflicts as you go through the commits.

If you are stuck, let me know and we can do this together!

**Comment by Luthaf** [comment]
This is now #255!

**Pull request closed without merging**